unison-cli/src/Unison/Cli/Monad.hs

```diff
@@ -178,10 +186,9 @@ data Env = Env
 --
 -- There's an additional pseudo @"currentPath"@ field lens, for convenience.
 data LoopState = LoopState
-  { root :: TMVar (Branch IO),
-    lastSavedRootHash :: CausalHash,
-    -- the current position in the namespace
-    currentPathStack :: List.NonEmpty Path.Absolute,
+  { currentProjectRoot :: TMVar (Branch IO),
```

Comment:
TODO: Can I remove the in-memory root entirely and just look it up from the codebase branch cache every time?
This would prevent any possible sync-mismatch errors 🤞🏼